game/world/objects/npc.cpp

```diff
@@ -2818,16 +2818,24 @@ void Npc::runEffect(Effect&& e) {
 bool Npc::isTargetableBySpell(TargetType t) const {
   if(bool(t&(TARGET_TYPE_ALL|TARGET_TYPE_NPCS)))
     return true;
+  bool g2 = owner.version().game==2;
   Guild gil = Guild(trueGuild());
   if(bool(t&TARGET_TYPE_HUMANS) && gil<GIL_SEPERATOR_HUM)
     return true;
-  if(bool(t&TARGET_TYPE_ORCS) && gil>GIL_SEPERATOR_ORC)
-    return true;
+  if(bool(t&TARGET_TYPE_ORCS)) {
+    if(g2 && gil>GIL_SEPERATOR_ORC)
```

Comment:
Nitpick:
Can we do, for sake of readability something like:
```c++
const bool g2 = owner.version().game==2;
const auto SEPERATOR_ORC = g2 ? GIL_SEPERATOR_ORC : GIL_G1_SEPERATOR_ORC;
... // other separators
```
and then keep most of original code - otherwise `if-else` tower becomes too complex.

Reply:
I've adapted the change for the separator. Question would be, do you also want to do the undead-thing differently? Because there it isn't a single seperator-value but comparing to a bunch of ids.

We could of course do
```
const auto G1_UNDEAD = (gil == GIL_G1_ZOMBIE || ...)
const auto G2_UNDEAD = (gil == GIL_GOBBO_SKELETON || ...)
...
if(bool(t&TARGET_TYPE_UNDEAD) && !g2 && G1_UNDEAD)
  return true;
if(bool(t&TARGET_TYPE_UNDEAD) && g2 && G2_UNDEAD)
  return true;
```
That would even shave one level off the if-tree :-)

Reply:
I was thinking about:
`const auto SKELETON = g2 ? GIL_SKELETON : GIL_G1_SKELETON`

However your solution looks even better, considering goblin-skeletons.

Reply:
Ok thanks ... I'll adapt the change accordingly :-)

Meanwhile I also found the other SEPERATOR_ORC occurence in Npc::isMonster(), that I missed earlier

Reply:
I think the updated patches from just now might do it :-) .

Not sure how you're convention is for resolving review comments. Do resolve it as done or do you resolve them as being satisfied? I know projects doing it either way, so just want to find out what you prefer.

Reply:
> Not sure how your convention is for resolving review comments.

No strict rules, for me it's better to resolve them myself - make it easier to follow

Reply:
> > Not sure how your convention is for resolving review comments.
> 
> No strict rules, for me it's better to resolve them myself - make it easier to follow

Good to know - also matches how we do it at my work - so don't even need to switch gears :-D